modules/image-loading-optimization/module.php

```diff
@@ -0,0 +1,346 @@
+<?php
+namespace Elementor\Modules\ImageLoadingOptimization;
+
+use Elementor\Core\Base\Module as BaseModule;
+use Elementor\Core\Experiments\Manager as Experiments_Manager;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly
+}
+
+class Module extends BaseModule {
+	/**
+	 * @var string The experiment name.
+	 */
+	const EXPERIMENT_NAME = 'e_image_loading_optimization';
+
+	/**
+	 * @var int Minimum square-pixels threshold.
+	 */
+	private $min_priority_img_pixels = 50000;
+
+	/**
+	 * @var int The number of content media elements to not lazy-load.
+	 */
+	private $omit_threshold = 3;
+
+	/**
+	 * @var array Keep a track of images for which loading optimization strategy were computed.
+	 */
+	private static $image_visited = [];
+
+	/**
+	 * Get Module name.
+	 */
+	public function get_name() {
+		return 'image-loading-optimization';
+	}
+
+	/**
+	 * Get experimental data.
+	 *
+	 * @return array Experimental settings.
+	 */
+	public static function get_experimental_data() {
+		return [
+			'name' => static::EXPERIMENT_NAME,
+			'title' => esc_html__( 'Optimize Image Loading', 'elementor' ),
+			'tag' => esc_html__( 'Performance', 'elementor' ),
+			'description' => esc_html__( 'Applying `fetchpriority="high"` on LCP image and lazy loading on images below the fold.', 'elementor' ),
+			'release_status' => Experiments_Manager::RELEASE_STATUS_ALPHA,
+			'default' => Experiments_Manager::STATE_INACTIVE,
+		];
+	}
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		parent::__construct();
+
+		// Stop wp core logic.
+		add_action( 'init', [ $this, 'stop_core_fetchpriority_high_logic' ] );
+		add_filter( 'wp_lazy_loading_enabled', '__return_false' );
```

Comment:
If we explicitly set this to false here, I am wondering if there might be any negative side-effects for 3rd-party plugins that use this?
![Screenshot 2023-09-26 at 16 24 10](https://github.com/elementor/elementor/assets/20996103/2b0c8cb4-8c18-43ca-9f63-712539e166d2)

Reply:
Good question. We disable it here because we're completely replacing the functionality, but let me consider a way that we could support this core flag and still over load the functionality that depends on it.